File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int s=0;
        for(int i=0;i<nums.size();i++){
            if((nums[i]^nums[i+1])==0){
            s=nums[i];
            break;}
        }
        vector<int>n;
        n.push_back(s);
        int k=0;
        for (int i=1;i<=nums.size();i++){
            k+=i;
        }
        int l=0;
        for (int j=0;j<nums.size();j++){
            if(nums[j]!=s){
                l+=nums[j];
            }
        }
        int m=k-(l+s);
        n.push_back(m);
        
        
        return n;
        
    }
};

File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        nums.sort()
        c=0
        for i in range(len(nums)):
            if(i!=len(nums)-1):
                if((nums[i]^nums[i+1])==0):
                    c=nums[i]
                    break
            else:
                break


            
        l=[]
        l.append(c)
        r=0
        for i in nums:
            if(i!=c):
                r+=i
        j=0
        for i in range(1,len(nums)+1):
            j+=i
        k=j-(r+c)
        l.append(k)
        return l
            



        